service/header/subscription_test.go

```diff
@@ -54,9 +50,33 @@ func TestSubscriber(t *testing.T) {
 	err = p2pSub2.Start(context.Background())
 	require.NoError(t, err)
 
+	done := make(chan bool, 1)
+	go func() {
+		for {
+			if net.Hosts()[1].Peerstore().Peers().Len() == 2 && net.Hosts()[0].Peerstore().Peers().Len() == 2 {
+				close(done)
+				break
+			}
+			time.Sleep(100 * time.Millisecond)
+		}
+	}()
+	ticker := time.NewTicker(2 * time.Second)
```

Comment:
Do we need to tick periodcally? time.Timer ?

Reply:
D'oh